src/util/tseries.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <algorithm>
+#include <cstdint>
+#include <cstring>
+#include <vector>
+
+#include "types.h"
+
+namespace bpftrace::util {
+
+template <typename T>
+std::pair<T, uint64_t> reduce_tseries_value(const std::vector<uint8_t> &values,
+                                            int nvalues,
+                                            const SizedType &inner_type)
+{
+  // Combine values only from the same epoch and return the reduced value from
+  // the latest epoch in this bucket.
+  std::map<uint64_t, std::pair<T, uint64_t>> epoch_to_value;
+  uint64_t latest_epoch = 0;
+
+  for (int i = 0; i < nvalues; i++) {
+    const uint8_t *val = values.data() + (i * sizeof(uint64_t) * 3);
```

Comment:
`3` is a bit "magical". Can we derive this some other way that makes it clear it's the size of the tseries struct and makes it work if the size changes later?